shared/protocol/src/main/java/io/pravega/shared/NameUtils.java

```diff
@@ -184,6 +199,22 @@ public static String getTransactionNameFromId(String parentStreamSegmentName, UU
         return result.toString();
     }
 
+    /**
+     * Returns the transient name for a TransientSegment based on the name of the current Parent StreamSegment, and the transientId.
+     *
+     * @param parentStreamSegmentName The name of the Parent StreamSegment for this transient segment.
+     * @param transientId             The unique Id for the transient segment.
+     * @return The name of the Transient StreamSegmentId.
+     */
+    public static String getTransientNameFromId(String parentStreamSegmentName, UUID transientId) {
```

Comment:
Did you add a unit test to `SegmentToContainerMapper` to validate that the transient segment name will map to the exact same container id as its parent name? There is code in it to do that for transaction segments; we need to make sure the same holds for transient segments too.